ros2_rust_galactic.repos

```diff
@@ -11,8 +11,11 @@ repositories:
     type: git
     url: https://github.com/ros2/rosidl_defaults.git
     version: galactic
-  # ros2-rust/ros2_rust:
-  #   type: git
-  #   url: https://github.com/ros2-rust/ros2_rust.git
-  #   version: master
-
+  colcon/colcon-cargo:
```

Comment:
I wouldn't put this in a `.repos` file, but just use `pip` to install the Python package

Reply:
I'm not sure how to make that work. Can you give more detail? If we don't have colcon-cargo checked out, it would need to be published on PyPI, right?

Reply:
No problem! You can install Python packages using `pip` directly from Git repositories `pip install git+https://github.com/colcon/colcon-cargo.git` should be sufficient. You can even pass a tag or a commit id like this `pip install git+https://github.com/colcon/colcon-ros-cargo.git@v1`

We should tag `colcon-cargo` so you can more easily depend on it.

Reply:
Yes

Reply:
Thanks for clarifying. This works with transitive dependencies? And the resulting packages still need to be built by colcon, right?

From my current point of view, I'd favor the `.repos` approach, since Python package management has a deserved reputation for being really finicky. We're using `vcs` anyway, so there's no extra cognitive load. Everything is visible and in one place (the ros2_rust dir).

Differently stated, what does using `pip` here buy us?

Reply:
> This works with transitive dependencies?

Yes.

> And the resulting packages still need to be built by colcon, right?

Not sure I'm following. `colcon` plugins are just Python packages, so once they are in the PYTHONPATH, `colcon` will discover them and use them, but they are not built by `colcon`.

> From my current point of view, I'd favor the .repos approach, since Python package management has a deserved reputation for being really finicky. We're using vcs anyway, so there's no extra cognitive load. Everything is visible and in one place (the ros2_rust dir).

Eventually that won't be the case, `colcon` plugins are released independently (e.g. `colcon-ros` is just a Python package that's been published to PyPI https://pypi.org/project/colcon-ros/)

> Differently stated, what does using pip here buy us?

`pip` is the standard way of installing Python packages, it can also use PyPI (the Python equivalent to crates.io), and it's the way all the other `colcon` plugins are installed.

Reply:
Ok, so if I understand right, we wouldn't need the two-stage build process anymore (if we forget about cargo-ros-install for a second). That's a nice benefit.

Now about cargo-ros-install, I could upload it to crates.io and we install it from there. Otherwise, if there's a way to execute the `cargo build` command as part of the setup process of colcon-ros-cargo (which is invoked by pip, right?), then that would also solve the issue. Can you run custom commands during setup, à la build.rs?

Reply:
I wouldn't make `cargo build` a part of the install process of `colcon-ros-cargo`. `pip` may be invoked as root and not the as the user that has `cargo` installed, it may be installed into a virtualenv, etc. If you want to be very strict with ROS dependencies, I'd create a `colcon-ros-install_vendor` package that checks if the `colcon-ros-install` crate is installed, and if not, run `cargo build/install`. But I'd just push `colcon-ros-install` to crates.io and add a line in the README telling users to run `cargo install colcon-ros-install`

Reply:
Ok. I'll probably rename it to `cargo-ros-build` or `cargo-ament-build`.

Reply:
Renamed to `cargo-ament-build`. Can you rename the new repo?